src/assisted_test_infra/test_infra/controllers/node_controllers/aws_controller.py

```diff
@@ -0,0 +1,155 @@
+from assisted_test_infra.test_infra.controllers.node_controllers.node_controller import NodeController
+from assisted_test_infra.test_infra.helper_classes.config.base_aws_config import BaseAwsConfig
+from assisted_test_infra.test_infra.tools import terraform_utils
+from assisted_test_infra.test_infra.utils import TerraformControllerUtil, utils
+from service_client import log
+
+
+class AwsController(NodeController):
+    _config: BaseAwsConfig
+
+    def __init__(self, config: BaseAwsConfig, cluster_config: BaseClusterConfig):
+        super().__init__(config, cluster_config)
+        self.cluster_name = cluster_config.cluster_name.get()
+        folder = TerraformControllerUtil.create_folder(self.cluster_name, platform=config.tf_platform)
+        self._tf = terraform_utils.TerraformUtils(working_dir=folder, terraform_init=False)
+        self._aws_client = None
+
+    def prepare_nodes(self):
+        config = {**self._config.get_all(), **self._entity_config.get_all(), "cluster_name": self.cluster_name}
+        self._tf.set_and_apply(**config)
+
+        return self.list_nodes()
+
+    def _get_vm(self, node_name: str) -> Dict[str, Any]:
+        return next((vm for vm in self._get_tf_vms() if vm["attributes"]["tags"]["Name"] == node_name), None)
+
+    def _get_tf_vms(self) -> List[Dict[str, Any]]:
+        vms_object_type = self._tf.get_resources(resource_type="aws_instance")
+
+        if not vms_object_type:
+            return list()
+
+        return [vm_instance for vms_objects in vms_object_type for vm_instance in vms_objects["instances"]]
+
+    def _aws_vm_to_node(self, terraform_vm_state: Dict[str, Any]) -> Node:
+        return Node(
+            name=terraform_vm_state["attributes"]["tags"]["Name"],
+            private_ssh_key_path=None,
+            node_controller=self,
+        )
+
+    def list_nodes(self) -> List[Node]:
+        tf_vms = self._get_tf_vms()
+        return list(map(self._aws_vm_to_node, tf_vms))
+
+    def get_cpu_cores(self, node_name: str) -> int:
+        return self._get_vm(node_name)["attributes"]["cpu_core_count"]
+
+    def get_ram_kib(self, node_name: str) -> int:
+        return 16 * 10**9
+
+    def get_node_ips_and_macs(self, node_name) -> Tuple[List[str], List[str]]:
+        ips = [self._get_vm(node_name)["attributes"]["private_ip"]]
+        return ips, ["DE:AD:BE:EF:DE:AD"]
+
+    def destroy_all_nodes(self) -> None:
+        self._tf.destroy(force=False)
+
+    def start_node(self, node_name: str, check_ips: bool) -> None:
+        pass
+
+    def start_all_nodes(self) -> List[Node]:
+        raise NotImplementedError
+
+    def shutdown_node(self, node_name: str) -> None:
+        raise NotImplementedError
+
+    def shutdown_all_nodes(self) -> None:
+        raise NotImplementedError
+
+    def restart_node(self, node_name: str) -> None:
+        raise NotImplementedError
+
+    def get_ingress_and_api_vips(self):
+        return None
```

Comment:
@osherdp FYI returning `None` here triggered UMN, I did not had to set the none platform.